freestyle/shared/src/main/scala/freestyle/macroimpl/Util.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2017 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package freestyle.macroimpl
+
+import scala.collection.immutable.Seq
+import scala.meta._
+import scala.meta.Defn.{ Class, Trait, Object }
+
+/* Utilities for scalameta independents of freestyle*/
+private[this] object ScalametaUtil {
```

Comment:
If there's only one class/trait in the file, let's give them the same name. Right now `Util.scala` just has `object ScalametaUtil`.

Also, I prefer that we consistently abbreviate certain things. Below `type` is abbreviated as `t` and `ty` and left as `type` in some places-- it will be easier to read the code using these helpers if it's all consistent.